### ISSUE TITLE
Can we modify the code to train for binary weighted network and not XNOR?

### ISSUE THREAD

**Issue by 25b3nk** [jiecaoyu/XNOR-Net-PyTorch#34]
Hey @jiecaoyu , first of all appreciate the work. Great job! :+1: 
When I check the code I only saw binarization of convolution parameters (weights). Is it XNOR or BWN?
Please let me know what I might be missing here.
If it binarizes the input also, can you point to where it does, so that I can edit and try training BWN?

Edit:
Hey @jiecaoyu , got to know where you binarize the input to the layer! It is using the function
`BinActive()`
Now if I just remove the function, will it be BWN or is there something else I need to do?

**Comment by jiecaoyu** [comment]
Hi @csbenk it will be BWN, I haven't tried running that so I am not sure whether current hyperparameters can help achieve the best accuracy.

**Comment by 25b3nk** [comment]
Thanks for the response @jiecaoyu , the BWN training for MNIST and CIFAR-10 worked with same hyperparametes. I could achieve 90% on CIFAR-10. Will try to validate the results and then post the results.
For imagenet training, I have not gone through the data but you have mentioned that the function for reading data has not been committed. Can I mail you for the same?

**Comment by jiecaoyu** [comment]
Hi @csbenk , please follow my answer [here](https://github.com/jiecaoyu/XNOR-Net-PyTorch/issues/6). Thanks!

**Comment by 25b3nk** [comment]
Thanks a lot @jiecaoyu ! :+1:  :pray:
